lib/usd/hdMaya/adapters/proxyAdapter.cpp

```diff
@@ -155,7 +149,9 @@ void HdMayaProxyAdapter::CreateUsdImagingDelegate() {
         _id.AppendChild(TfToken(TfStringPrintf(
             "ProxyDelegate_%s_%p", _proxy->name().asChar(), _proxy))),
         _proxy, GetDagPath()));
-    _isPopulated = false;
+
+    _usdDelegate->Populate(stage->GetPseudoRoot());
+    _isPopulated = true;
```

Comment:
I don't like the side effect of calling `_usdDelegate->Populate` inside of `CreateUsdImagingDelegate()`. Would it be possible to instead call `HdMayaProxyAdapter::Populate()` from `_OnStageSet` after the `CreateUsdImagingDelegate` call there? The code right now is confusing because both orderings of calls is safe (`Populate` can automatically call `CreateUsdImagingDelegate` & `CreateUsdImagingDelegate` can call `Populate`).

I guess the real question is why have two methods if you can call them in either order and get the right answer? And if the two separate methods are different enough to really be necessary, are they named correctly to describe what they actually do?